vllm/v1/utils.py

```diff
@@ -101,7 +101,16 @@ def __init__(
         target_fn: Callable,
         process_kwargs: dict[Any, Any],
     ):
-        context = get_mp_context()
+        assert "executor_class" in process_kwargs, "executor_class is missing"
+
+        override_mp_method, override_reason = None, None
+        if process_kwargs["executor_class"].uses_ray:
```

Comment:
it's not specific to ray though. it's whether the current process is a ray actor. if it is, we need to use spawn anyway.

Reply:
Not really. Even we are in a ray actor we can still use fork with multiprocessing executor, so the issue is only for the Ray executor.

Reply:
> Even we are in a ray actor we can still use fork with multiprocessing executor

it works right now, but forking in ray actor is an anti-pattern and it might break something else.